core/src/components/router/utils/path.ts

```diff
@@ -9,6 +10,26 @@ export const generatePath = (segments: string[]): string => {
   return '/' + path;
 };
 
+const generateUrl = (segments: string[], useHash: boolean, queryString?: string) => {
```

Comment:
Can we add some tests for `generateUrl`?

Reply:
I would make a similar comment as I did for `removePrefix`:

- I would prefer not to add an `export` for test only,
- The function is tested via `writePath`.